packages/amplify-app/src/index.js

```diff
@@ -8,8 +8,12 @@ const frameworkConfigMapping = require('./framework-config-mapping');
 const args = require('yargs').argv;
 const { addFileToXcodeProj } = require('./xcodeHelpers');
 const ini = require('ini');
+const semver = require('semver');
+const stripAnsi = require('strip-ansi');
 
 const amplifyCliPackageName = '@aws-amplify/cli@canary';
```

Comment:
Please remove @canary, as we don't want to have +1 PR just to remove it.

We should do version check the NPM way, put the minimum versions into ```packages.json``` and read it from there:
```
  "engines": {
    "node": ">=8.12.0",
    "@aws-amplify/cli": ">=3.17.1"
  },
```

Also the version we should check should be 4.0.0 not 3.17.1